### PR TITLE
Apm 2521 back channel logout endpoint

### PR DESCRIPTION
**Pull request by jslb**

### CODE REVIEW

File: e2e/tests/conftest.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ async def get_token_cis2_token_exchange(
     token_resp = await oauth.get_token_response(
         grant_type="token_exchange",
         data={
-            "grant_type": "urn:ietf:params:oauth:grant-type:token-exchange",
+            "grant_type": "urn:etf:params:oeuth:grant-type:token-exchange",
```
Comment (openfinch) [inline e2e/tests/conftest.py:123]: Who did this? Why did they do this?

File: proxies/live/apiproxy/policies/RaiseFault.BadRequest.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<RaiseFault async="false" continueOnError="false" enabled="true" name="RaiseFault.BadRequest">
```
Comment (openfinch) [inline proxies/live/apiproxy/policies/RaiseFault.BadRequest.xml:1]: Check if this is stilled being used.

File: e2e/tests/conftest.py [hunk 3]
```diff
@@ -277,7 +277,7 @@ def setup_session(request):
     print("\nCreating Default App..")
     asyncio.run(
         app.create_new_app(
-            callback_url="https://nhsd-apim-testing-internal-dev.herokuapp.com/callback"
+            callback_url="https://httpbin.org/anything"
```
Comment (openfinch) [inline e2e/tests/conftest.py:280]: Can probably revert this now.

File: e2e/tests/oauth/conftest.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import pytest
```
Comment (openfinch) [inline e2e/tests/oauth/conftest.py:1]: Are we using any of this?

File: azure/azure-pr-pipeline.yml [hunk 5]
```diff
@@ -32,13 +32,11 @@ extends:
     service_name: ${{ variables.service_name }}
     short_service_name: ${{ variables.short_service_name }}
     service_base_path: ${{ variables.service_base_path }}
-    product_display_name: ${{ variables.product_display_name }}
-    product_description: ${{ variables.product_description }}
     ping: false
     jinja_templates:
-      IDENTITY_PROVIDER_CIS2: cis2
+      IDENTITY_PROVIDER_CIS2: cis2-keycloak
```
Comment (openfinch) [inline azure/azure-pr-pipeline.yml:37]: Can we revert this?

File: e2e/tests/oauth/test_backchannel_logout.py [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+            'client_id': self.oauth.client_id,
+            'redirect_uri': 'https://httpbin.org/anything',
+            'response_type': 'code',
+            'state': str(uuid4())
+        }.items()])
+        authorize_url = f"{self.oauth.base_uri}/authorize?{params}"
+
+        driver.get(authorize_url)
+        username = driver.find_element(By.ID, 'username')
+        username.send_keys('aal3' + Keys.ENTER)
```
Comment (openfinch) [inline e2e/tests/oauth/test_backchannel_logout.py:133]: This should be an nhs-number username

File: e2e/tests/oauth/utils/authenticator.py [hunk 7]
```diff
@@ -0,0 +1,79 @@
+import e2e.tests.oauth.utils.config as config
```
Comment (openfinch) [inline e2e/tests/oauth/utils/authenticator.py:1]: Is this needed?

File: e2e/tests/oauth/utils/check_oauth.py [hunk 8]
```diff
@@ -0,0 +1,36 @@
+import requests
```
Comment (openfinch) [inline e2e/tests/oauth/utils/check_oauth.py:1]: Is this needed?

File: e2e/tests/oauth/utils/config.py [hunk 9]
```diff
@@ -0,0 +1,40 @@
+from e2e.tests.oauth.utils.environment import EnvVarWrapper
```
Comment (openfinch) [inline e2e/tests/oauth/utils/config.py:1]: Is this needed?

File: e2e/tests/oauth/utils/environment.py [hunk 10]
```diff
@@ -0,0 +1,24 @@
+import os
```
Comment (openfinch) [inline e2e/tests/oauth/utils/environment.py:1]: Is this needed?

**Review by openfinch — DISMISSED** [review]

**Review by strutt — APPROVED** [review]

**Review by strutt — APPROVED** [review]

**Pull request merged**